### DISCUSSION TITLE
Next 14 app directory error Select

### DISCUSSION THREAD

**Discussion by goodlythink** [mantinedev/mantine#5478] [Q&A]
my Select code

      <Select
        searchable
        clearable
        placeholder="ค้นหาตามจังหวัด"
        data={provinces.data.map((field: any) => ({ value: field.name, label: field.name }))}
        value={!!customerStore.province.value === false ? null : customerStore.province.value}
        onChange={handleChangeProvince}
      />
      
 When refresh page it's error

Error: Hydration failed because the initial UI does not match what was rendered on the server.

Warning: Expected server HTML to contain a matching <button> in <div>.

See more info here: https://nextjs.org/docs/messages/react-hydration-error

but if i change code on value to

      <Select
        searchable
        clearable
        placeholder="ค้นหาตามจังหวัด"
        data={provinces.data.map((field: any) => ({ value: field.name, label: field.name }))}
        value="country"
        onChange={handleChangeProvince}
      />

it not error on refresh page'

how to fix please.

**Comment by rtivital** [comment]
If you read value from browser storage, move this logic to useEffect

**Reply by goodlythink** [reply]
thank you. it's work

But I don't understand. I don't read values from the browser storage, but rather read values from the store obtained from Zustand. On the contrary, if you use this method in React, there shouldn't be any issues.

//in normal react not error
value={!!customerStore.province.value === false ? null : customerStore.province.value}
